### DISCUSSION TITLE
Simple question: router or not?

### DISCUSSION THREAD

**Discussion by fede843** [Koenkk/zigbee2mqtt#19255] [Devices]
Hi, how do I know if a device works as router or just an end device?
For instance: https://www.zigbee2mqtt.io/devices/TS0601_dimmer_1.html
It does not say a word about it. Is there a place to check that? DB does not say anything neither https://zigbee.blakadder.com/Moes_MS-105Z.html

**Comment by Koenkk** [comment]
This info is indeed not available on the site, but generally speaking you can assume that any mains powered device is a router and a battery powered device is an enddevice

**Comment by fede843** [comment]
Yes as a thumb rule you can say that, but as you said, it is based in assumption and it is not always the case.
For people in the other extreme of the world, who need to wait 3 or more months, pay double the price, have a fight in customs to release the item, and then find out it is not router, it is far away from convenient.

Can we make a proper feature request? Many items are in the market since a while and a vast number of users could contribute for the cause. I think this is beneficial for everyone. It could be just a field at least, saying "router/end device/not available yet".

**Reply by Koenkk** [reply]
Yes you can submit a feature request.

**Comment by fede843** [comment]
Found a related issue: https://github.com/blakadder/zigbee/issues/1051

**Comment by fede843** [comment]
sorry for my ignorance, probably a silly question, but how are you guys gathering all the available fields, data, etc for all the devices?

**Reply by fede843** [reply]
https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html

**Comment by francisp2** [comment]
I would add: assume anything 'no neutral' is not a router.

**Comment by fede843** [comment]
Created the feature request https://github.com/Koenkk/zigbee2mqtt/issues/19268
Closing this one.
Thanks!
